src/test/java/de/tum/in/www1/artemis/competency/StandardizedCompetencyIntegrationTest.java

```diff
@@ -46,71 +47,69 @@ class StandardizedCompetencyIntegrationTest extends AbstractSpringIntegrationLoc
     @Autowired
     private SourceRepository sourceRepository;
 
-    static class CheckStandardizedCompetencyValidationProvider implements ArgumentsProvider {
+    @Autowired
+    private UserUtilService userUtilService;
 
-        @Override
-        public Stream<Arguments> provideArguments(ExtensionContext extensionContext) {
-            var competency1 = new StandardizedCompetency("  ", "valid description", CompetencyTaxonomy.ANALYZE, null);
-            var competency2 = new StandardizedCompetency("0".repeat(StandardizedCompetency.MAX_TITLE_LENGTH + 1), "valid description", CompetencyTaxonomy.ANALYZE, null);
-            var competency3 = new StandardizedCompetency("valid title", "0".repeat(StandardizedCompetency.MAX_DESCRIPTION_LENGTH + 1), CompetencyTaxonomy.ANALYZE, null);
-            return Stream.of(Arguments.of(competency1), Arguments.of(competency2), Arguments.of(competency3));
-        }
-    }
+    @Autowired
+    private StandardizedCompetencyUtilService standardizedCompetencyUtilService;
 
     @BeforeEach
     void setupTestScenario() {
         userUtilService.addUsers(TEST_PREFIX, 1, 1, 1, 1);
 
-        knowledgeArea = new KnowledgeArea("Knowledge Area 0", "KA0", "KA description");
-        knowledgeArea = knowledgeAreaRepository.save(knowledgeArea);
-
-        standardizedCompetency = new StandardizedCompetency("Competency 0", "SC description", CompetencyTaxonomy.ANALYZE, "1.0.0");
-        standardizedCompetency.setKnowledgeArea(knowledgeArea);
-        standardizedCompetency = standardizedCompetencyRepository.save(standardizedCompetency);
+        knowledgeArea = standardizedCompetencyUtilService.saveKnowledgeArea("Knowledge Area 0", "KA0", "KA description", null);
+        standardizedCompetency = standardizedCompetencyUtilService.saveStandardizedCompetency("Competency 0", "SC description", CompetencyTaxonomy.ANALYZE, "1.0.0", knowledgeArea,
+                null);
 
-        source = new Source("Source 0", "Author 0", "http:localhost:8000");
+        source = new Source("Source 0", "Author 0", "localhost:8000");
         source = sourceRepository.save(source);
     }
 
-    private void testAllPreAuthorizeInstructor() throws Exception {
-        request.get("/api/standardized-competencies/1", HttpStatus.FORBIDDEN, StandardizedCompetency.class);
-        request.get("/api/standardized-competencies/for-tree-view", HttpStatus.FORBIDDEN, StandardizedCompetency.class);
-        request.get("/api/standardized-competencies/knowledge-areas/1", HttpStatus.FORBIDDEN, KnowledgeArea.class);
-    }
+    @Nested
+    class PreAuthorize {
```

Comment:
Seems like unnecessary checks to me. This is gated behind checks that ensure that /admin endpoints are annotated with the correct enforcement annotation and v.v.

Reply:
That is the point, yes 🤷‍♂️ IMO it's worth to check for developer errors to ensure that the security does not get compromised

Reply:
The thing I'm saying is - there are already checks in place. I you would put an @EnforceAdmin annotation on a non admin path - a test would fail. If you would put another annotation than the @EnforceAdmin annotation on an admin path - a test would fail. The /admin endpoints are also double protected as they are also checked against the admin role in the SecurityConfiguration.

So if you only want to check for developer errors, IMO the checks are already done. If you want to check for a developer removing/changing these checks and then suddenly everything is possible, we could test for that but I don't think this is reasonable to test for ^^

Reply:
Good point, I'll remove the tests =)